### DISCUSSION TITLE
Creating a MAUI class library

### DISCUSSION THREAD

**Discussion by solomonfried** [dotnet/maui#12043] [Q&A]
I have been having an issue trying to publish a UWP app that contains a MAUI class library.
 (https://github.com/microsoft/microsoft-ui-xaml/issues/7846#issuecomment-1347563662)

While waiting for that to be resolved I had a question regarding MAUI class libraries. 
In Xamarin, if I wanted a shared library that included Controls (Pages, Views...), I would create a Net Standard 2 library and add Xamarin.Forms Nuget package to it. I could then Add ContentPages, ContentViews. etc... into that library.

With MAUI, it seems I need to create a "MAUI Class Library" to accomplish the same thing. If I create a simple .NET 6 library, I am unable to add Pages, Buttons, etc...

Am I correct, or am I doing something wrong?

Thank you

**Comment by Xyncgas** [comment]
Sounds about right because a regular class library has no idea what you mean by MAUI, just like with blazor or platform specific projects, a regular class library just isn't aware of the api they provide.

For example, when you are writing on a blank piece of paper, it doesn't know how to open up the camera on IOS.

If you are developing an algorithm for let's say compression, that would be used on these devices / platforms / projects, then you can use a simple class library and write the code on a blank file because it's just math.

Hope it helps.

Merry Christmas.

**Comment by xtuzy** [comment]
When use 'usemaui' tag in class library, it will auto refer maui. It like xamarinform refer nuget packages.

**Reply by solomonfried** [reply]
> When use 'usemaui' tag in class library, it will auto refer maui. It like xamarinform refer nuget packages.

Thanks for the suggestion.
Seems to allow me to add Views and other Maui controls, but not quite ready....
On build, I get an error on InitializeComponent in a ContentView.
Also when going to Project, Add... Maui components are not offered. If I add a ContentView it is a Xamarin ContentView.

Thanks
